### PR TITLE
BWS-91: * worker return with array permissions

### PR DESCRIPTION
**Pull request by bohdanshcherba**

### CODE REVIEW

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 1]
```diff
@@ -44,6 +52,29 @@ class Worker {
...
+
+    const permissionIds = new Set<string>([]);
+
+    await Promise.all(
+      groupsIds.map(async (item) => {
+        const permissions = await this.#GroupsPermissionsModel
+          .query()
+          .select('permissionId')
+          .where({ 'groupId': item.groupId });
+        permissions.map((item) => {
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]: map should always return something

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 2]
```diff
@@ -44,6 +52,29 @@ class Worker {
...
+  public async getWorkerPermissions(workerId: string): Promise<Array<string>> {
+    const groupsIds = await this.#UsersGroupsModel
+      .query()
+      .select('groupId')
+      .where({ 'userId': workerId });
+
+    const permissionIds = new Set<string>([]);
+
+    await Promise.all(
+      groupsIds.map(async (item) => {
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]: use reduce instead

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 3]
```diff
@@ -44,6 +52,32 @@ class Worker {
...
+      .query()
+      .select('permissionId')
+      .whereIn('groupId', groupsIds);
+
+    const permissionIds = new Set<string>([]);
+
+    permissions.map((item) => {
+      permissionIds.add(item.permissionId);
+      return item;
+    });
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]:
```suggestion
    const permissionIds = permissions.reduce((acc, it) => acc.add(it), new Set());
```

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 4]
```diff
@@ -44,6 +52,32 @@ class Worker {
     return workers;
   }
 
+  public async getWorkerPermissions(workerId: string): Promise<Array<string>> {
+    type userGroups = {
+      groupId: string;
+    };
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]:
```suggestion
```

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 5]
```diff
@@ -44,6 +52,32 @@ class Worker {
     return workers;
   }
 
+  public async getWorkerPermissions(workerId: string): Promise<Array<string>> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]:
```suggestion
  public async getWorkerPermissions(workerId: string): Promise<string[]> {
```

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 6]
```diff
@@ -44,6 +52,32 @@ class Worker {
...
+      .whereIn('groupId', groupsIds);
+
+    const permissionIds = new Set<string>([]);
+
+    permissions.map((item) => {
+      permissionIds.add(item.permissionId);
+      return item;
+    });
+
+    return Array.from(permissionIds);
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]: do we need to return a name?

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/services/worker/worker.entity.ts [hunk 7]
```diff
@@ -7,6 +8,7 @@ class Worker {
   public passwordSalt: string;
   public tenantId: string;
   public groupIds: string[];
+  public permissions: EAMPermissionGetAllItemResponseDto[] | null;
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.entity.ts]:
```suggestion
  public permissions: string[];
```
should be an array of string (names)

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/services/worker/worker.entity.ts (hunk 7 above)
Comment (bohdanshcherba) [reply backend/src/services/worker/worker.entity.ts]: but workers can be without group and permission will be null

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 8]
```diff
@@ -44,6 +58,42 @@ class Worker {
     return workers;
   }
 
+  public async getWorkerPermissions(
+    workerId: string,
+  ): Promise<string[] | null> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]:
```suggestion
  ): Promise<string[]> {
```

File: backend/src/data/repositories/worker/worker.repository.ts [hunk 9]
```diff
@@ -44,6 +58,42 @@ class Worker {
...
+    const groups = await this.#UsersGroupsModel
+      .query()
+      .select('groupId')
+      .where({ 'userId': workerId });
+
+    if (!groups) {
+      return null;
```
Comment (what1s1ove) [inline backend/src/data/repositories/worker/worker.repository.ts]:
```suggestion
      return [];
```

File: backend/src/services/worker/worker.entity.ts [hunk 10]
```diff
@@ -7,6 +7,7 @@ class Worker {
   public passwordSalt: string;
   public tenantId: string;
   public groupIds: string[];
+  public permissions: string[] | null;
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.entity.ts]:
```suggestion
  public permissions: string[];
```

File: backend/src/services/worker/worker.entity.ts [hunk 11]
```diff
@@ -50,6 +55,7 @@ class Worker {
     passwordSalt: string;
     tenantId: string;
     groupIds: string[];
+    permissions: string[] | null;
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.entity.ts]:
```suggestion
    permissions: string[];
```

File: backend/src/services/worker/worker.entity.ts [hunk 12]
```diff
@@ -24,6 +26,7 @@ class Worker {
...
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.entity.ts]:
```suggestion
    permissions: string[];
```

File: backend/src/services/worker/worker.entity.ts [hunk 13]
```diff
@@ -67,12 +74,14 @@ class Worker {
...
     passwordSalt,
     groupIds,
+    permissions,
     tenantId,
   }: {
     name: string;
     passwordHash: string;
     passwordSalt: string;
     groupIds: string[];
+    permissions: string[] | null;
```
Comment (what1s1ove) [inline backend/src/services/worker/worker.entity.ts]:
```suggestion
    permissions: string[];
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**